pkg/backup/item_collector.go

```diff
@@ -164,7 +181,7 @@ func getOrderedResourcesForType(orderedResources map[string]string, resourceType
 }
 
 // getResourceItems collects all relevant items for a given group-version-resource.
-func (r *itemCollector) getResourceItems(log logrus.FieldLogger, gv schema.GroupVersion, resource metav1.APIResource) ([]*kubernetesResource, error) {
+func (r *itemCollector) getResourceItems(log logrus.FieldLogger, gv schema.GroupVersion, resource metav1.APIResource, resourceIDsMap map[schema.GroupResource][]velero.ResourceIdentifier) ([]*kubernetesResource, error) {
```

Comment:
So the issue here is that for finalize, we don't need to deal with include/exclude since we have a limited list of items that we know we need to include, so instead of include/exclude rules plus listing items from cluster, we pull items from the map. We still need the rest of this logic since it does all of the GVR and preferred version negotiation which is necessary for getting items in their proper path for the tarball. But I can add clearer comments to the func to make this distinction clear. There are other places in velero code where we skip include/exclude rules under certain contexts.